### PR TITLE
Include missing voyage filters

### PR DESCRIPTION
**Pull request by migueluzcategui**
#### RELATED TICKETS

- https://vortexa.atlassian.net/browse/RND-6724

#### RELATED PRs

- NA

#### CHANGELOG

- Implement the filters that we currently have at api level in the SDK, voyage_date_activity, events_types, origin and destination behaviours.

#### TESTS

- describe your test strategy

#### COMMENTS

- any other business

### CODE REVIEW

File: tests/endpoints/test_voyages_timeseries.py [hunk 1]
```diff
@@ -11,13 +11,66 @@ class TestVoyagesTimeseries(TestCaseUsingRealAPI):
...
+                time_min=start,
+                time_max=end,
+                origins=rotterdam,
+                voyage_date_range_activity="departures",
+            )
+            .to_df()
+        )
+        assert len(df) >= numbers_of_days_between_start_and_end
+
+    def test_search_departures_with_last_discharge_behavuour(self):
```
Comment (SeanBarry) [inline tests/endpoints/test_voyages_timeseries.py:57]: typo -> behaviour

File: vortexasdk/endpoints/voyages_congestion_breakdown.py [hunk 2]
```diff
@@ -210,6 +214,12 @@ def search(
             order_direction: Determines the direction of sorting. ‘asc’ for ascending, ‘desc’ for
             descending.
 
+            voyage_date_range_activity: Filter to determinate the mode into how the voyages should be counted,`active`, `departures` or `arrivals`.
+
+            origin_behaviour: The origin behaviour into how the voyage_date_range_activity should consider in departure mode, it could be `first_load` or `any_load`.
+
+            destination_behaviour: The destination behaviour into how the voyage_date_range_activity should consider in arrival mode, it could be `last_discharge` or `any_discharge`.
+
```
Comment (kahoowkh) [inline vortexasdk/endpoints/voyages_congestion_breakdown.py:222]: Is `intra_movements` missing here?

File: vortexasdk/endpoints/voyages_congestion_breakdown.py [hunk 3]
```diff
@@ -83,6 +83,10 @@ def search(
         vessel_risk_level_excluded: Union[str, List[str]] = None,
         has_ship_to_ship: str = None,
         has_charterer: str = None,
+        intra_movements: str = None,
```
Comment (SeanBarry) [inline vortexasdk/endpoints/voyages_congestion_breakdown.py:86]: These type annotations need to be updated.
A str can't be None by default

File: vortexasdk/endpoints/voyages_congestion_breakdown.py [hunk 4]
```diff
@@ -210,6 +214,12 @@ def search(
             order_direction: Determines the direction of sorting. ‘asc’ for ascending, ‘desc’ for
             descending.
 
+            voyage_date_range_activity: Filter to determinate the mode into how the voyages should be counted,`active`, `departures` or `arrivals`.
```
Comment (SeanBarry) [inline vortexasdk/endpoints/voyages_congestion_breakdown.py:217]: typo -> determine how the voyages

File: vortexasdk/endpoints/voyages_geography_breakdown.py [hunk 5]
```diff
@@ -114,6 +114,9 @@ def search(
         time_charterer: Union[ID, List[ID]] = None,
         time_charterer_excluded: Union[ID, List[ID]] = None,
         intra_movements: str = None,
+        voyage_date_range_activity: str = None,
```
Comment (SeanBarry) [inline vortexasdk/endpoints/voyages_geography_breakdown.py:117]: Same type issue as above, can you also fix the other occurrences of it?

File: vortexasdk/endpoints/voyages_congestion_breakdown.py [hunk 6]
```diff
@@ -210,6 +214,12 @@ def search(
...
+
+            origin_behaviour: The origin behaviour into how the voyage_date_range_activity should consider in departure mode, it could be `first_load` or `any_load`.
```
Comment (SeanBarry) [inline vortexasdk/endpoints/voyages_congestion_breakdown.py:219]: This is hard to understand, I think it needs to be rephrased:

```origin_behaviour: The origin behaviour determines which departure mode the `voyage_date_range_activity` should count, must be one of  [`first_load`, `any_load`].```

File: vortexasdk/endpoints/voyages_congestion_breakdown.py [hunk 7]
```diff
@@ -210,6 +214,12 @@ def search(
...
+
+            destination_behaviour: The destination behaviour into how the voyage_date_range_activity should consider in arrival mode, it could be `last_discharge` or `any_discharge`.
```
Comment (SeanBarry) [inline vortexasdk/endpoints/voyages_congestion_breakdown.py:221]: This is hard to understand, I think it needs to be rephrased:

```destination_behaviour: The destination behaviour determines which arrival mode the `voyage_date_range_activity` should count, must be one of  [`last_discharge`, `any_discharge`].```

File: vortexasdk/endpoints/voyages_geography_breakdown.py [hunk 8]
```diff
@@ -251,6 +254,12 @@ def search(
 
             intra_movements: Filter movements based on whether the vessel started and ended in the same country, or geographical layer.
 
+            voyage_date_range_activity: Filter to determinate the mode into how the voyages should be counted,`active`, `departures` or `arrivals`.
```
Comment (SeanBarry) [inline vortexasdk/endpoints/voyages_geography_breakdown.py:257]: Suggested reprhase: 
```Filter to determine how the voyages should be counted. Must be one of [`active`, `departures`, `arrivals`]```

File: vortexasdk/endpoints/voyages_geography_breakdown.py [hunk 9]
```diff
@@ -251,6 +254,12 @@ def search(
...
+
+            origin_behaviour: The origin behaviour into how the voyage_date_range_activity should consider in departure mode, it could be `first_load` or `any_load`.
+
+            destination_behaviour: The destination behaviour into how the voyage_date_range_activity should consider in arrival mode, it could be `last_discharge` or `any_discharge`.
```
Comment (SeanBarry) [inline vortexasdk/endpoints/voyages_geography_breakdown.py:261]: I'd suggest the same rephrase here as I suggested earlier in the PR.

File: vortexasdk/endpoints/voyages_product_breakdown.py [hunk 10]
```diff
@@ -250,6 +253,12 @@ def search(
 
             intra_movements: Filter movements based on whether the vessel started and ended in the same country, or geographical layer.
 
+            voyage_date_range_activity: Filter to determinate the mode into how the voyages should be counted,`active`, `departures` or `arrivals`.
+
+            origin_behaviour: The origin behaviour into how the voyage_date_range_activity should consider in departure mode, it could be `first_load` or `any_load`.
+
+            destination_behaviour: The destination behaviour into how the voyage_date_range_activity should consider in arrival mode, it could be `last_discharge` or `any_discharge`.
+
```
Comment (SeanBarry) [inline vortexasdk/endpoints/voyages_product_breakdown.py:261]: Same rephrases suggested here

File: vortexasdk/endpoints/voyages_routes_breakdown.py [hunk 11]
```diff
@@ -250,6 +253,12 @@ def search(
 
             intra_movements: Filter movements based on whether the vessel started and ended in the same country, or geographical layer.
 
+            voyage_date_range_activity: Filter to determinate the mode into how the voyages should be counted,`active`, `departures` or `arrivals`.
+
+            origin_behaviour: The origin behaviour into how the voyage_date_range_activity should consider in departure mode, it could be `first_load` or `any_load`.
+
+            destination_behaviour: The destination behaviour into how the voyage_date_range_activity should consider in arrival mode, it could be `last_discharge` or `any_discharge`.
+
```
Comment (SeanBarry) [inline vortexasdk/endpoints/voyages_routes_breakdown.py:261]: Same rephrases suggested here

**Review by SeanBarry — CHANGES_REQUESTED** [review]
I've suggested some rephrasing that would need to be implemented in many places. I haven't commented on every single place.

The current phrasing is hard to understand.

I've also suggested some type fixes, I know these type errors are present already but it's important we all understand why they are wrong, and don't introduce new ones.

File: vortexasdk/endpoints/voyages_congestion_breakdown.py (hunk 3 above)
Comment (migueluzcategui) [reply vortexasdk/endpoints/voyages_congestion_breakdown.py:91]: I think the current implementation of the PythonSDK is expecting for all these optional fields to be by default `None` so the payload sent to the APIs is just what the client is explicit asking and the defaults are handle at API level

File: vortexasdk/endpoints/voyages_congestion_breakdown.py (hunk 2 above)
Comment (migueluzcategui) [reply vortexasdk/endpoints/voyages_congestion_breakdown.py:229]: You're right it was missing

File: vortexasdk/endpoints/voyages_geography_breakdown.py (hunk 5 above)
Comment (migueluzcategui) [reply vortexasdk/endpoints/voyages_geography_breakdown.py]: Changed to use a enum instead

File: vortexasdk/endpoints/voyages_congestion_breakdown.py (hunk 6 above)
Comment (migueluzcategui) [reply vortexasdk/endpoints/voyages_congestion_breakdown.py]: changed too

File: vortexasdk/endpoints/voyages_congestion_breakdown.py (hunk 7 above)
Comment (migueluzcategui) [reply vortexasdk/endpoints/voyages_congestion_breakdown.py]: changed

File: vortexasdk/endpoints/voyages_geography_breakdown.py (hunk 8 above)
Comment (migueluzcategui) [reply vortexasdk/endpoints/voyages_geography_breakdown.py]: changed

File: vortexasdk/endpoints/voyages_geography_breakdown.py [hunk 12]
```diff
@@ -251,6 +254,12 @@ def search(
 
             intra_movements: Filter movements based on whether the vessel started and ended in the same country, or geographical layer.
 
+            voyage_date_range_activity: Filter to determinate the mode into how the voyages should be counted,`active`, `departures` or `arrivals`.
+
+            origin_behaviour: The origin behaviour into how the voyage_date_range_activity should consider in departure mode, it could be `first_load` or `any_load`.
```
[the remark this answers is not in the document]
Comment (migueluzcategui) [reply vortexasdk/endpoints/voyages_geography_breakdown.py]: changed

File: vortexasdk/endpoints/voyages_product_breakdown.py (hunk 10 above)
Comment (migueluzcategui) [reply vortexasdk/endpoints/voyages_product_breakdown.py]: changed

File: vortexasdk/endpoints/voyages_congestion_breakdown.py (hunk 3 above)
Comment (SeanBarry) [reply vortexasdk/endpoints/voyages_congestion_breakdown.py:91]: That's fine, we can keep them as None. The point is that the type annotation is impossible (I am aware this is a problem all throughout the SDK but wanted to raise awareness so we all stop increasing the number of occurrences of this problem).
We can't type something as `str` then initialise it to `None`. It should be typed as `str | None`!

**Review by SeanBarry — APPROVED** [review]

**Pull request merged**